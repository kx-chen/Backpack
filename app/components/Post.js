import styled from 'styled-components';
import React from 'react';
import PropTypes from 'prop-types';

const PostContainer = styled.div`
  display: flex;
  background-color: white;
  color: black;
  margin-top: 30px;
  width: 750px;
  border-radius: 4px;
  border: 1px solid #ccc;
  cursor: pointer;

  :hover {
    border: 1px solid #898989;
  }
`;

const PostLinkAndTitle = styled.a`
  font-weight: 600;
  font-size: 20px;
`;

const UpvotesContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  min-width: 50px;
  background-color: rgba(135, 138, 140, 0.2);
  text-align: center;
  font-weight: 600;
`;

const PostBodyContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px;
`;

const PostAuthorAndPostingDate = styled.span`
  font-size: 13px;
  color: rgb(120, 124, 126);
`;

const CommentsIndicator = styled.span`
  color: #878a8c;
  font-weight: 600;
  margin-top: 10px;
  font-size: 13px;
`;

Post.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  numComments: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  // created_utc: PropTypes.string.isRequired,
  ups: PropTypes.string.isRequired,
  onPostSelected: PropTypes.func.isRequired
};
// author_flair_text
function Post({ id, title, numComments, author, ups, onPostSelected }) {
  return (
    <PostContainer onClick={() => onPostSelected(id)}>
      <UpvotesContainer>{ups}</UpvotesContainer>
      <PostBodyContainer>
        <PostAuthorAndPostingDate>
          Posted by u/{author}
        </PostAuthorAndPostingDate>
        <PostLinkAndTitle>{title}</PostLinkAndTitle>
        <CommentsIndicator>{numComments} Comments</CommentsIndicator>
      </PostBodyContainer>
    </PostContainer>
  );
}

export default Post;
