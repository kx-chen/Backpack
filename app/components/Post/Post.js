// @flow
import React, { Component } from 'react';
import Col from 'react-bootstrap/Col';
import { connect } from 'react-redux';
import styles from '../../containers/App/App.css';

type Props = {
  title: string,
  karma: string
};

class Post extends Component<Props> {
  props: Props;

  render() {
    const { title, karma } = this.props;
    console.log('Post props', this.props);
    return (
      <Col className={styles.col}>
        {title}
        {karma}
      </Col>
    );
  }
}

function mapStateToProps(state) {
  return {
    title: state.postData.title,
    karma: state.postData.karma
  };
}

export default connect(mapStateToProps)(Post);
