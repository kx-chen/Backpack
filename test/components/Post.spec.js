import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Post from '../../app/components/Post';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const actions = {
    onPostSelected: spy()
  };
  const component = shallow(<Post id={'abce'}
                                  key={'abcd'}
                                  title={'efg'}
                                  numComments={'23'}
                                  author={'Dog'}
                                  ups={'39'}
                                  {...actions}/>);
  return {
    component,
    actions,
    onPostSelected: component.find('#post-container'),
  };
}

describe('Post component', () => {
  it('should call onPostSelected on click', () => {
    const { onPostSelected, actions } = setup();
    onPostSelected.simulate('click');
    expect(actions.onPostSelected.called).toBe(true);
  });

  it('should match exact snapshot', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
