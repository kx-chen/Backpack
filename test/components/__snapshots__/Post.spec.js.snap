// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should match exact snapshot 1`] = `ShallowWrapper {}`;
